packages/block-library/src/heading/style.scss

```diff
@@ -7,4 +7,8 @@ h6 {
 	&.has-background {
 		padding: $block-bg-padding--v $block-bg-padding--h;
 	}
+	&.has-text-align-right[style*="writing-mode:vertical-rl"],
+	&.has-text-align-left[style*="writing-mode:vertical-lr"] {
```

Comment:
We could use the two selectors everywhere, I was just trying to not have a selector that will never be used.
It may be better to have two selectors, until (if ever) the heading block has other styles in editor.scss.

Reply:
I deleted my previous comment because I did not read the code correctly :)
It is working! Thank you.